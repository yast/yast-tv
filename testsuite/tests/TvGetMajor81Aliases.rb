# encoding: utf-8

module Yast
  class TvGetMajor81AliasesClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # none alias
      @READ_alias_nil = { "modules" => { "alias" => nil } }

      # no char-major-81 alias
      @READ_no_alias_81 = {
        "modules" => { "alias" => { "eth0" => "", "fb0" => "", "midi" => "" } }
      }

      # one char-major-81 alias
      @READ_one_alias_81 = {
        "modules" => {
          "alias" => { "eth0" => "", "fb0" => "", "char-major-81-0" => "" }
        }
      }

      # more char-major-81 aliases
      @READ_more_aliases_81 = {
        "modules" => {
          "alias" => {
            "eth0"            => "",
            "char-major-81-0" => "",
            "char-major-81-1" => "",
            "midi"            => ""
          }
        }
      }


      TEST(lambda { Tv.GetMajor81Aliases(path(".modules")) }, [
        @READ_alias_nil,
        {},
        {}
      ], false)

      TEST(lambda { Tv.GetMajor81Aliases(path(".modules")) }, [
        @READ_no_alias_81,
        {},
        {}
      ], false)

      TEST(lambda { Tv.GetMajor81Aliases(path(".modules")) }, [
        @READ_one_alias_81,
        {},
        {}
      ], false)

      TEST(lambda { Tv.GetMajor81Aliases(path(".modules")) }, [
        @READ_more_aliases_81,
        {},
        {}
      ], false)

      nil
    end
  end
end

Yast::TvGetMajor81AliasesClient.new.main
