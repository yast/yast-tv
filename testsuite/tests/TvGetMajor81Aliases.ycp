{
    // testedfiles: Tv.ycp

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    // none alias
    map READ_alias_nil =
	$[ "modules" :
	    $[ "alias" : nil ]
	];

    // no char-major-81 alias
    map READ_no_alias_81 =
	$[ "modules" :
	    $[ "alias" :
		$[ "eth0" : "" , "fb0" : "", "midi" : "" ]
	    ]
	];

    // one char-major-81 alias
    map READ_one_alias_81 =
	$[ "modules" :
	    $[ "alias" :
		$[ "eth0" : "", "fb0" : "", "char-major-81-0" : "" ]
	    ]
	];

    // more char-major-81 aliases
    map READ_more_aliases_81 =
	$[ "modules" :
	    $[ "alias" :
		$[ "eth0" : "", "char-major-81-0" : "", "char-major-81-1" : "", "midi" : "" ] 
	    ]
	];


    TEST(``(Tv::GetMajor81Aliases()), [ READ_alias_nil , $[], $[] ], false);

    TEST(``(Tv::GetMajor81Aliases()), [ READ_no_alias_81 , $[], $[] ], false);

    TEST(``(Tv::GetMajor81Aliases()), [ READ_one_alias_81 , $[], $[] ], false);

    TEST(``(Tv::GetMajor81Aliases()), [ READ_more_aliases_81 , $[], $[] ], false);
}
