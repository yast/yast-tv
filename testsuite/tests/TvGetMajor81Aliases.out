Read	.target.tmpdir "/tmp"
Read	.target.size "/etc/install.inf" -1
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Dir	.modules.alias: []
Return	[]
Dir	.modules.alias: ["eth0", "fb0", "midi"]
Return	[]
Dir	.modules.alias: ["char-major-81-0", "eth0", "fb0"]
Return	["char-major-81-0"]
Dir	.modules.alias: ["char-major-81-0", "char-major-81-1", "eth0", "midi"]
Return	["char-major-81-0", "char-major-81-1"]
